mne_bids/read.py

```diff
@@ -328,7 +328,117 @@ def _handle_channels_reading(channels_fname, bids_fname, raw):
     return raw
 
 
-def read_raw_bids(bids_fname, bids_root, extra_params=None,
+def _infer_kind(*, bids_basename, bids_root, sub, ses):
+    # Check which kind is available for this particular
+    # subject & session. If we get no or multiple hits, throw an error.
+
+    kinds = _get_kinds_for_sub(bids_basename=bids_basename,
+                               bids_root=bids_root, sub=sub, ses=ses)
+    if not kinds:
+        raise ValueError('No data found.')
+    elif len(kinds) >= 2:
+        msg = (f'Found data of more than one recording modality. Please '
+               f'pass the `kind` parameter to specify which data to load. '
+               f'Found the following kinds: {kinds}')
+        raise RuntimeError(msg)
+    elif len(kinds) == 1:
+        kind = kinds[0]
+
+    return kind
+
+
+def _get_bids_fname_from_filesystem(*, bids_basename, bids_root, sub, ses,
```

Comment:
what is this * ? :) never seen it before

Reply:
Keyword-only arguments :)

Reply:
Okay with you? I like that because it makes refactoring easier and is more explicit. Also prefer this in parts of public APIs – but that's something for a future discussion :) 

https://www.python.org/dev/peps/pep-3102/

Reply:
example for it being used in a public API: https://github.com/xdf-modules/xdf-python/pull/59

Reply:
Yep. We can / should discuss introducing this across the public API. Sometime later. ;)

Reply:
In my personal projects, I've been using this for a long time, see e.g. https://github.com/hoechenberger/questplus/blob/master/questplus/qp.py#L10-L20

Reply:
I think it's a nice trick, specially for private functions because we often mess up when passing arguments around

Reply:
For public API, I wouldn't like to be forced to use keyword arguments if I was just passing one or two parameters. Maybe using the * after first 2 or 3 params is fine.